source/entities/Creature.cpp

```diff
@@ -559,6 +519,7 @@ void Creature::importFromPacket(ODPacket& is)
         mWeaponR = getGameMap()->getWeapon(tempString);
         OD_ASSERT_TRUE_MSG(mWeaponR != nullptr, "Unknown weapon name=" + tempString);
     }
+    setupDefinition(*getGameMap(), *ConfigManager::getSingleton().getCreatureDefinitionDefaultWorker());
```

Comment:
I misread. You already don't call setupDefinition in importFromStream. It would be nice to avoid this call here but I don't think it is possible...